dbt_semantic_interfaces/protocols/export.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import Optional, Protocol
+
+from dbt_semantic_interfaces.type_enums.export_destination_type import (
+    ExportDestinationType,
+)
+
+
+class Export(Protocol):
+    """Configuration for writing query results to a table."""
+
+    @property
+    @abstractmethod
+    def name(self) -> str:  # noqa: D
+        pass
+
+    @property
+    @abstractmethod
+    def config(self) -> ExportConfig:  # noqa: D
+        pass
+
+
+class ExportConfig(Protocol):
+    """Nested configuration attributes for exports."""
+
+    @property
+    @abstractmethod
+    def export_as(self) -> ExportDestinationType:
+        """Type of destination to write export to."""
+        pass
+
+    @property
+    @abstractmethod
+    def schema_name(self) -> Optional[str]:
```

Comment:
@QMalcolm the core parser can take `schema` from yaml and convert it to schema_name just as we do with Pydantic here, correct? I'm assuming that's the case.

Reply:
That is indeed the case. We have a separate unparsed and parsed node representation. It'll be `schema` in the unparsed and `schema_nam`e in the parsed